crates/core_arch/src/arm/neon.rs

```diff
@@ -1136,12 +1388,526 @@ pub unsafe fn vtbx4_p8(a: poly8x8_t, b: poly8x8x4_t, c: uint8x8_t) -> poly8x8_t
     ))
 }
 
+macro_rules! arm_vget_lane {
+    ($name:ident, $from:ty, $to:ty, $lanes:literal) => {
+        #[inline]
+        #[target_feature(enable = "neon")]
+        #[cfg_attr(target_arch = "arm", target_feature(enable = "v7"))]
+        #[cfg_attr(test, assert_instr(umov))]
```

Comment:
Missing `#[rustc_args_required_const(1)]` ?

Reply:
This is a oddity, if the argument to this function is const it compiles to to a `mov` instead of a `umov`. The whole `get_lane` thing is driving me mildly insane. 

Same as strictly the line `arm_vget_lane!(vgetq_lane_u64, uint64x2_t, u64, 2);` should be `arm_vget_lane!(vgetq_lane_u64, uint64x2_t, u64, 1);` (as it's either 0 or 1 not 0, 1, or 2 bug again using the correct `1` there it compiles to a incorrect opcode.

Reply:
The actual argument of this function in C is `__constrange(0,1) int` and per the arm reference it should lower to a mov for `arm32+v7` and to `umov` for `aarch64`, so the `mov` looks correct to me here.

Reply:
I couldn't find that information on the specs, can you link me to where to find that? All I got is 
 [this](https://developer.arm.com/docs/ddi0596/a/a64-simd-and-floating-point-instructions-alphabetic-order/umov-unsigned-move-vector-element-to-general-purpose-register) saying umov but something about the alias conditions is that what we're hitting?

Reply:
The signatures of the arm intrinsics headers can be found in the ARM info center (e.g. http://infocenter.arm.com/help/index.jsp?topic=/com.arm.doc.dui0491c/BABJFCGC.html) along with the instruction they lower to. 

The signature in the https://developer.arm.com/architectures/instruction-sets/simd-isas/neon/intrinsics?search=vgetq_lane_u64 also says that this argument is a compile-time constant and therefore #[rustc_args_required_const(1)] must be used, but only the instruction this lowers to on A64 is shown there.

Reply:
So updating it to this:
```rust
/// Move vector element to general-purpose register
#[inline]
#[target_feature(enable = "neon")]
#[cfg_attr(target_arch = "arm", target_feature(enable = "v7"))]
#[rustc_args_required_const(1)]
#[cfg_attr(all(test, target_arch = "arm"), assert_instr(mov, imm5 = 1))]
#[cfg_attr(all(test, target_arch = "aarch64"), assert_instr(vmov, imm5 = 1))]
pub unsafe fn vgetq_lane_u64(v: uint64x2_t, imm5: i32) -> u64 {
    if (imm5) < 0 || (imm5) > 1 {
        unreachable_unchecked()
    }
    let imm5 = (imm5 & 0b1) as u32;
    simd_extract(v, imm5)
}
```
which works for the others still gives a mov even on aarch64. The macro gets around the const by wrapping it in a case and then using a constant value to pass into the simd_extract but at least it generates the `vmov` instruction given by the manual. But I'm impartial to the approach - rewriting it as a function gets rid of a macro that's a win.

Reply:
I updated it to the above code and selected `mov` for both `arm` and `aarch64` linking to this ticket for refference.